src/service/data_products/common_functions.py

```diff
@@ -268,16 +269,20 @@ async def query_simple_collection_list(
         async for d in cur:
             if not filters.count:
                 if not filters.match_spec.is_null_subset():
-                    d[match_field] = filters.match_spec.get_prefixed_subset_id() in d[
-                        names.FLD_MATCHES_SELECTIONS]
+                    d[match_field] = _get_matchsel(filters.match_spec, d)
                 if not filters.selection_spec.is_null_subset():
-                    d[selection_field] = filters.selection_spec.get_prefixed_subset_id() in d[
-                        names.FLD_MATCHES_SELECTIONS]
+                    d[selection_field] = _get_matchsel(filters.selection_spec, d)
             acceptor(d)
     finally:
         await cur.close(ignore_missing=True)
 
 
+def _get_matchsel(spec: SubsetSpecification, doc: dict[str, Any]):
+    if not doc.get(names.FLD_MATCHES_SELECTIONS):
+        return False
```

Comment:
not `list()` but `False` here?

Reply:
No, `__match__` and `__sel__` field values are expected to be `True` or `False`

Reply:
I added the missing return type to make this clearer

Reply:
👍